Stop worker if connection is closed,

BaillyKappa opened the pull request:
…bk fork

Description of PR...

## Changes

- Item 1
- Item 2

## API Updates

### New Features *(required)*

This section should include details regarding new features added to the library.
This should include both open and private APIs. If including both, create two
sub headers underneath this one to label updates accordingly.

### Deprecations *(required)*

All deprecations should be listed here in order to ensure the reviewer understands
which sections of the codebase will affect contributors and users of the library.

### Enhancements *(optional)*

The enhancements section should include code updates that were included with this
pull request. This sections should detail refactoring that might have affected
other parts of the library.

## Checklist

- [ ] Unit tests
- [ ] Documentation

## References

(optional)

Include __important__ links regarding the implementation of this PR.
This usually includes and RFC or an aggregation of issues and/or individual conversations
that helped put this solution together. This helps ensure there is a good aggregation
of resources regarding the implementation.

Fixes #85, Fixes #22, Fixes username/repo#123
Connects #123

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/camunda-community-hub/pyzeebe?pullRequest=612) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/camunda-community-hub/pyzeebe?pullRequest=612) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/camunda-community-hub/pyzeebe?pullRequest=612) it.</sub>

BaillyKappa commented:
Hi @dimastbk, I have re-created your PR #526 and included my fix to stop worker.work() if the connection to zeebe is lost since your initial changes were not sufficient to solve the issue I and @matthieu-dubin bumped into (see PR #526 comments). 
Additionally, I have added a small note in the documentation regarding this change. Any chance you can add yourself as reviewer and we try to merge it?

Sorry for the multiple commits I was testing the fix on my side and also fixed the poetry lint errors.

btw if you want to use my fork @matthieu-dubin you can use git+https://github.com/BaillyKappa/pyzeebe.git in your requirements.txt file, I am using it on my current project. This way I am able to have the health check of my pod to fail and the pod to restart after the 10 retries if we lose the connection to the zeebe gateway.

BaillyKappa commented:
Hi @dimastbk, do you think we can merge the PR since all the UTs and CTs are good now? Or let me know if I need to change anything else.

dimastbk commented:
Hi. Maybe we should use only callbacks or monitoring tasks? I think they have doubled functionals.

BaillyKappa commented:
Hi, indeed after paying more attention to the code you are right but I remember when I tested it with the callback it was not doing anything after the retries even if it should. I will try to re-do some tests this week with the callback.

BaillyKappa commented:
I removed my monitor_connection function and fix the callback issue that I had.         self._stop_event = anyio.Event() was duplicated in my code, that's why my first tests failed. Now it works.
